leaderelection/src/main/java/io/scalecube/leaderelection/LeadershipEvent.java

```diff
@@ -0,0 +1,27 @@
+package io.scalecube.leaderelection;
+
+/**
+ * Event which is emitted on cluster leadership changes when member state changed from FOLLOWER, CANDIDATE, LEADER.
+ * 
+ * @author Ronen Nachmias
+ *
+ */
+public class LeadershipEvent {
+
+  private final StateType fromState;
+  private final StateType toState;

```

Comment:
I think this type of event not really interesting for public API - it's more like internal state transition. I think that public event should be like, about changes of leader - new leader elected, you get leadership, you lost leadership. Also it's not clear from/to state is related to local member only?  How will I get event about changing in leadership if not local member was elected as leader.